arxiv/base/urls/links.py

```diff
@@ -176,6 +177,53 @@ def _this_url_text(attrs: Attrs, new: bool = False) -> Attrs:
     return attrs
 
 
+def _remove_scheme_from_label(attrs: Attrs, new: bool = False) -> Attrs:
+    label = attrs['_text']
+    o = urlparse(label)
+    if o.scheme and f'{o.scheme}://' in label:
+        _, attrs['_text'] = label.split(f'{o.scheme}://', 1)
+    return attrs
+
+
+def _shorten_text_to_30_characters(attrs: Attrs, new: bool = False) -> Attrs:
```

Comment:
I would rename it with something like `_shorten_url_to_n_characters` because it is not for any text, but a more specific type (i.e., a URL) and encoding an arbitrary number in the function name means if the limit is to be changes in future, all the function references will need a modification. This function can have an extra parameter of `max_length` which can default to `30` or whatever number we settle with.

Reply:
Renamed to ``_shorten_label_to_n_characters`` in c2348a9. Because of the way this is used (as a callback for bleach) we need to keep the parameterization via the config proxy. In other words, an app can override the default by setting ``MAX_URL_DISPLAY_LENGTH``.